### PR TITLE
DS-1100 Migrate schedules fix to main repo

### PR DESCRIPTION
**Pull request by froboy**
@podarok if you create a new branch, we can merge this into the source repo on a separate branch, then it's not as fragile a connection.

from https://github.com/seattleymca/openy_repeat/tree/YGSSLA-396

### CODE REVIEW

**Comment by rollki** [comment]
@froboy 
For one of the clients, we can observe a problem with duplicate categories in the left side.
image

Actuallyб the problem is wrong content management, as someone created 2 or more categories (activity CT) with the same title (one of which with an empty subcategory field) that's why the query returns 2 records with the same title, and later vuejs renders it and perceives it as one and the same, since the keys are different, but the values are the same.

Theoretically, the problem can be solved on the side of content management, but it is probably better to use distinct and prevent such cases.

I don't think we can solve both problems, as we can only solve the technical side.

Solution:
`$query->distinct(TRUE);`

Screenshot 2023-12-20 at 15 08 27

**Comment by rollki** [comment]
@froboy 
If you agree with the decision, I propose to do it in this PR :) 
But we can discuss it of course

**Comment by froboy** [comment]
@rollki sorry I missed this. I've added that fix.

File: src/Plugin/Block/RepeatSchedulesBlock.php [hunk 1]
```diff
@@ -121,17 +121,15 @@ public function getLocations() {
...
-    $query->leftJoin('node__field_activity_category', 'sc', 'sc.entity_id = nfd.nid');
-    $query->condition('sc.field_activity_category_target_id', $config->get('activity'));
     $query->fields('nfd', ['title']);
     $query->condition('type', 'activity');
     $query->condition('status', 1);
     if ($nids) {
       $query->condition('nid', $nids, 'NOT IN');
     }
     $query->orderBy('title');
+    $query->distinct(TRUE);
```
Comment (podarok) [inline src/Plugin/Block/RepeatSchedulesBlock.php:132]: would be great to have Drupal->logger->warning("We have duplicates here, please do cleanup action")

Comment (froboy) [reply src/Plugin/Block/RepeatSchedulesBlock.php:132]: @podarok we're not currently testing if there are duplicates or not, and the `distinct` is set without a condition. I think what you're suggesting would require some additional code like:

```
$query->execute()
if 
  $results have duplicates
then 
  log->warning
  $query->distinct
  $query->execute
```

That seems possibly overkill to me?

**Comment by podarok** [comment]
https://github.com/ynorth-projects/openy_repeat/tree/schedules_fix @froboy

**Pull request merged**